### PR TITLE
Index settings and mappings

### PR DESCRIPTION
**Pull request by eksoverzero**
Please advise if this makes sense and what else, besides tests, you need.

### CODE REVIEW

File: src/connection.js [hunk 1]
```diff
@@ -33,12 +35,24 @@ var Connection = function( options, callback ) {
...
   this.client.get( {
     index: this._index,
     type: params.type,
     id: params.id
   }, ( error, response ) => {
-    if( error && error.displayName !== 'NotFound' ) {
+    if( error && response.error && response.error.type == 'index_not_found_exception' ) {
+      var body = { settings: {}, mappings: {} }
+
+      body.settings = JSON.parse(this._settings)
```
Comment (yasserf) [inline src/connection.js]: Since settings don't change, might make sense to do parse within constructor for as small optmization

**Comment by yasserf** [comment]
Since these are optional it makes perfect sense. Tests should be fine. Changes to any connectors or core components do need the cla to be signed tho, hope that is okay.

https://deepstream.io/info/community/cla/

**Comment by eksoverzero** [comment]
Working on your suggestion I found that the previous way I set the mappings and settings limited the mappings and settings to be applied to one type.
CLA done.

**Comment by yasserf** [comment]
Sorry for getting back so late. This looks good! Merging in

**Pull request merged**